### PR TITLE
Commit composer.lock to the repository

### PR DESCRIPTION
**Pull request by opengeek**
### What does it do?
Adds composer.lock to the project to ensure consistent installation of dependencies.

### Why is it needed?
To ensure consistent installation of dependencies when installing from git.

### Related issue(s)/PR(s)
#14814

### CODE REVIEW

**Comment by Mark-H** [comment]
It doesn't have to go into the build/distro, does it?

**Comment by opengeek** [comment]
I would think it should. At the very least the composer.json needs to be there, and it wasn't.

File: .gitignore [hunk 1]
```diff
@@ -49,7 +49,7 @@ _build/templates/default/sass/font-awesome
 # ignore composer files
 composer.phar
 #composer.json
-composer.lock
+#composer.lock
```
Comment (JoshuaLuckers) [inline .gitignore]: Maybe we can just remove the lines instead?

**Review by JoshuaLuckers — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -10,7 +10,6 @@ php:
   - 7.3
   - 7.2
   - 7.1
-  - 7.0
```
Comment (JoshuaLuckers) [inline .travis.yml]: Does this mean we make PHP 7.1 the minimum required version for MODX 3?

File: composer.json [hunk 3]
```diff
@@ -43,7 +43,7 @@
         "vendor-dir": "core/vendor"
     },
     "require": {
-        "php": ">=7.0",
+        "php": ">=7.1",
```
Comment (JoshuaLuckers) [inline composer.json]: Isn't it a better idea to have a separate commit/pr for the minimum required version bump? If 7.1 is the minimum required version we also have to change the version checks (for example in `manager/index.php`).

**Review by JoshuaLuckers — CHANGES_REQUESTED** [review]

File: composer.json (hunk 3 above)
Comment (opengeek) [reply composer.json]: I had to do this so it passed the checks. 7.0 will no longer pass the unit tests.

Comment (JoshuaLuckers) [reply composer.json]: Then we need to officially bump the minimum PHP version required for MODX 3.x to 7.1.

**Comment by JoshuaLuckers** [comment]
@opengeek can we move forward with getting composer.lock in the repo?

**Comment by opengeek** [comment]
Yeah, now 7.0 and 7.1 won't work with composer.lock committed. I'm going to say that we avoid committing this to the repo for now.

**Pull request closed without merging**